lib/icinga/checkable-check.cpp

```diff
@@ -514,6 +506,10 @@ void Checkable::ProcessCheckResult(const CheckResult::Ptr& cr, const MessageOrig
 			SetSuppressedNotifications(suppressed_types_after);
 		}
 	}
+
+	/* update reachability for child objects */
+	if (hardChange && !children.empty())
+		OnReachabilityChanged(this, cr, children, origin);
```

Comment:
Dependencies have an attribute `ignore_soft_states`: https://icinga.com/docs/icinga-2/latest/doc/09-object-types/#dependency

So a soft state change can also result in a change of reachability.